### DISCUSSION TITLE
Temperature and salinity nudging at the boundaries

### DISCUSSION THREAD

**Discussion by zi-hua-liu** [NOAA-GFDL/CEFI-regional-MOM6#49] [Q&A] [answered]
May I ask how to set up the model configuration for nudging on the boundaries? There are 4 open boundaries in my model (regional mom6-sis2-cobaltV2) and only temperature and salinity fields on the 4 boundaries need to be nudged. Below is the corresponding setup for the OBCs, but I am not sure how to add the nudging data files to the boundaries. Any comments are appreciated.

**OBC_SEGMENT_001 = "I=N,J=0:N,FLATHER,ORLANSKI,NUDGED,ORLANSKI_TAN,NUDGED_TAN" !                             
OBC_SEGMENT_001_VELOCITY_NUDGING_TIMESCALES = 1.0, 360.0 !   [days] default = 0.0**
                                _! Timescales in days for nudging along a segment, for inflow, then outflow.
                                ! Setting both to zero should behave like SIMPLE obcs for the baroclinic
                                ! velocities._
**OBC_SEGMENT_002 = "J=N,I=N:0,FLATHER,ORLANSKI,NUDGED,ORLANSKI_TAN,NUDGED_TAN" !                              
OBC_SEGMENT_002_VELOCITY_NUDGING_TIMESCALES = 1.0, 360.0 !   [days] default = 0.0                              
OBC_SEGMENT_003 = "I=0,J=N:0,FLATHER,ORLANSKI,NUDGED,ORLANSKI_TAN,NUDGED_TAN" !                              
OBC_SEGMENT_003_VELOCITY_NUDGING_TIMESCALES = 1.0, 360.0 !   [days] default = 0.0                              
OBC_SEGMENT_004 = "J=0,I=0:N,FLATHER,ORLANSKI,NUDGED,ORLANSKI_TAN,NUDGED_TAN" !                                
OBC_SEGMENT_004_VELOCITY_NUDGING_TIMESCALES = 1.0, 360.0 !   [days] default = 0.0**

**OBC_TRACER_RESERVOIR_LENGTH_SCALE_OUT = 3.0E+03 !   [m] default = 0.0**
                                _! An effective length scale for restoring the tracer concentration at the
                                ! boundaries to externally imposed values when the flow is exiting the domain._
**OBC_TRACER_RESERVOIR_LENGTH_SCALE_IN = 3.0E+03 !   [m] default = 0.0**
                                _! An effective length scale for restoring the tracer concentration at the
                                ! boundaries to values from the interior when the flow is entering the domain._

Do I just need to enlarge the thickness of the boundaries (currently a single layer), or are there any other parameters that need to be added or modified? Thanks!

**Comment by yichengt900** [comment, marked as answer]
Hi @zi-hua-liu,

Thanks for bringing this up. There are several ways to do nudging in MOM6, and I think the one @charliestock mentioned is the `SPONGE` option in MOM6. In short, you will need to add the following runtime parameters to your `MOM_override`:

```
#override SPONGE = True
#override SPONGE_UV = False
#override SPONGE_DAMPING_FILE = "damping_full_t_90d.nc"
#override SPONGE_IDAMP_VAR = "Idamp"
#override SPONGE_STATE_FILE = "glorys_sponge_monthly_bnd_1990.nc"
#override SPONGE_PTEMP_VAR = "thetao"
#override SPONGE_SALT_VAR = "so"
#override INTERPOLATE_SPONGE_TIME_SPACE = True
#override SPONGE_DATA_ONGRID = True
```

where `damping_full_t_90d.nc` is a file with the sponge damping rates (normally we use strong nudging along the boundary and decrease to 0 in the interior domain), and glorys_sponge_monthly_bnd_1990.nc contains the state to damp toward. You can check these [scripts](https://github.com/NOAA-GFDL/CEFI-regional-MOM6/tree/main/tools/sponge) to learn how to generate your own files for nudging. Feel free to let me know if you have any further questions.

**Reply by zi-hua-liu** [reply]
Thanks for your quick and detailed response! These are super helpful. It works.
